tensorflow/core/kernels/mkl_avgpooling_op.cc

```diff
@@ -487,45 +486,63 @@ class MklAvgPoolingOp : public MklPoolingForwardOpBase<T> {
         return;
       }
 
-      // If input is in Mkl layout, then just get the memory format from it
-      // directly, instead of using input data_format to AvgPool.
-      if (dnn_shape_input.IsMklTensor()) {
-        dnn_data_output.SetUsrMem(
-            output_dims_mkl_order,
-            static_cast<memory::format>(
-                dnn_data_input.GetUsrMemDesc().data.format));
+      // Get the input memory descriptor
+      memory::desc input_md = dnn_shape_input.IsMklTensor()
+          ? dnn_shape_input.GetMklLayout()
+          : memory::desc(TFShapeToMklDnnDimsInNCHW(input_tensor_shape,
+                this->data_format_tf_),
+                MklDnnType<T>(), this->data_format_mkldnn_);
+
+      // Get src/filter/stride/padding information
+      memory::dims src_dims = dnn_shape_input.IsMklTensor()
```

Comment:
Could you please put all these memory::dims in a struct and add a helper function PoolParamsToDims to populate them. Use it in the functions below as well as much as possible. There's a lot of duplicated boilerplate.

Reply:
Done by code change.